src/tests/conformance/handshake.rs

```diff
@@ -223,96 +191,84 @@ async fn reject_non_version_replies_to_version() {
+                )
                 .await
-            {
-                Ok(_) => {}
-                Err(err) if is_termination_error(&err) => return,
-                Err(err) => panic!("Unexpected error while sending version: {:?}", err),
-            }
-
-            // (4) receive `verack` in response to our `version`
-            match Message::read_from_stream(&mut stream).await {
-                Ok(message) => assert_matches!(message, Message::Verack),
-                Err(err) if is_termination_error(&err) => {}
-                Err(err) => panic!("Unexpected error while receiving verack: {:?}", err),
-            }
-        }));
+                .unwrap();
+
+            // Receiving a Verack indicates the non-version message was ignored.
+            let (_, verack) = synthetic_node.recv_message_timeout(TIMEOUT).await.unwrap();
+            assert_matches!(verack, Message::Verack);
+
+            // Gracefully shut down the synthetic node.
+            synthetic_node.shut_down();
+        });
+
+        handles.push(handle);
     }
 
-    node.start().await;
+    // Start the node instance with the initial peers.
+    node.initial_peers(listeners).start().await;
 
+    // Run each future to completion.
     for handle in handles {
         handle.await.unwrap();
     }
 
+    // Gracefully shut down the node.
     node.stop().await;
 }
 
 #[tokio::test]
-async fn reject_non_verack_replies_to_verack() {
+async fn ignore_non_verack_replies_to_verack() {
     // Conformance test 005.
     //
-    // The node rejects non-Verack message as a response to initial Verack it sent.
-    //
-    // Test procedure:
-    //  For each non-verack message,
-    //
-    //  1. Expect `Version`
-    //  2. Send `Version`
-    //  3. Expect `Verack`
-    //  4. Send test message
-    //  5. Expect `Reject(Invalid)`
-    //  6. Expect connection to be terminated
+    // The node ignores non-Verack message as a response to initial Verack it sent.
     //
-    // This test currently fails as neither Zebra nor ZCashd currently fully comply
-    // with this behaviour, so we may need to revise our expectations.
-    //
-    // ZCashd node eagerly sends messages before handshake has been concluded.
-    // Zebra node does not send Reject, but terminates the connection.
-    //
-    // TODO: confirm expected behaviour
+    // zebra: disconnects.
+    // zcashd: completes the handshake and sends a few messages handled by the filter.
 
     let genesis_block = Block::testnet_genesis();
     let block_hash = genesis_block.double_sha256().unwrap();
     let block_inv = Inv::new(vec![genesis_block.inv_hash()]);
     let block_loc = LocatorHashes::new(vec![block_hash], Hash::zeroed());
-    let mut test_messages = vec![
-        Message::Version(Version::new(new_local_addr(), new_local_addr())),
+    let test_messages = vec![
         Message::GetAddr,
         Message::MemPool,
+        Message::Verack,
```

Comment:
This probably shouldn't be here for this test case?

```suggestion
```

Reply:
Good catch!